streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -1668,6 +1669,33 @@ public void shouldReturnAdminInstanceID() {
         }
     }
 
+    @Test
+    public void shouldReturnProducerAndConsumerInstanceIds() {
+        prepareStreams();
+        prepareStreamThread(streamThreadOne, 1);
+        prepareStreamThread(streamThreadTwo, 2);
+        final Uuid mainConsumerInstanceId = Uuid.randomUuid();
+        final Uuid producerInstanceId = Uuid.randomUuid();
+        final KafkaFutureImpl<Uuid> consumerFuture = new KafkaFutureImpl<>();
+        final KafkaFutureImpl<Uuid> producerFuture = new KafkaFutureImpl<>();
+        consumerFuture.complete(mainConsumerInstanceId);
+        producerFuture.complete(producerInstanceId);
+        final Uuid adminInstanceId = Uuid.randomUuid();
+        adminClient.setClientInstanceId(adminInstanceId);
```

Comment:
Why do we need  `adminClient` and `adminInstanceId` - seems both are unused below.

Reply:
Test throws an exception when not set, I've added it to the assertions